internal/pkg/config/config.go

```diff
@@ -155,19 +142,76 @@ func (c *Config) access() *ucfg.Config {
 
 // Merge merges two configuration together.
 func (c *Config) Merge(from interface{}, opts ...interface{}) error {
-	ucfgOpts, err := getUcfgOptions(opts...)
+	ucfgOpts, _, err := getOptions(opts...)
 	if err != nil {
 		return err
 	}
 	return c.access().Merge(from, ucfgOpts...)
 }
 
 // ToMapStr takes the config and transform it into a map[string]interface{}
-func (c *Config) ToMapStr() (map[string]interface{}, error) {
+func (c *Config) ToMapStr(opts ...interface{}) (map[string]interface{}, error) {
+	if len(opts) == 0 {
+		opts = DefaultOptions
+	}
+	ucfgOpts, local, err := getOptions(opts...)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Just a thought - not sure if it makes sense or not: it looks like the constructor, `NewConfigFrom` takes `opts` and defaults them to `DefaultOptions`, just like we're doing here.  And then it also calls `getOptions` just like here.  Would it make sense to define the `Config` type as a struct and persist either `opts` and/or `ucfgOpts` and `local` as fields in there so we can just use them here?

Reply:
Fair enough, not for this PR.